### ISSUE TITLE
Override Labels format doesn't appear to match kubernetes qualification

### ISSUE THREAD

**Issue by knisbet** [Shopify/kubeaudit#212]
##### ISSUE TYPE
 - [x] Bug Report

# BUG REPORT

##### SUMMARY

Sorry, not sure if I'm missing something, but I ran into this today when attempting to use override labels for pod capabilities. It appears that the label format being used doesn't match the validation in place in apimachinery / client-go.

##### ENVIRONMENT
* Kubeaudit version: v0.6.0
* Kubeaudit install method: DIY-BUILD

##### STEPS TO REPRODUCE
On a k8s cluster:
1. kubectl edit pod/\<any pod\>
2. Add to metadata.labels `container.audit.kubernetes.io/coredns/allow-capability-NET_BIND_SERVICE: "coredns-dns-port"`

##### EXPECTED RESULTS

I expected the label format for overrides to be compatible with kubernetes.

##### ACTUAL RESULTS

kubectl will report the following error:
```
# * metadata.labels: Invalid value: "container.audit.kubernetes.io/coredns/allow-capability-NET_BIND_SERVICE": a qualified name must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]') with an optional DNS subdomain prefix and '/' (e.g. 'example.com/MyName')
```

##### ADDITIONAL INFORMATION
The validation logic that kubernetes accepts is available at https://github.com/kubernetes/apimachinery/blob/master/pkg/util/validation/validation.go#L42-L71 which appears to only allow for a single `/` character.

**Comment by nschhina** [comment]
The override we use for capabilites is `kubeaudit.allow.capability.chown: "true"`

cc: @klautcomputing

**Comment by klautcomputing** [comment]
@knisbet 
TLDR: `audit.kubernetes.io/pod/allow-capability-net-bind-service: "You reason here"`
Reasoning: You cannot have `_` in there because kubernetes doesn't like that, so what we did is replace all the `_` with `-` and decided to keep it all lower caps.

**Comment by knisbet** [comment]
@klautcomputing The issue isn't about the `_` or `-` character, it's that the validation logic only allows for a single `/` character. audit.kubernetes.io/pod/allow-capability-net-bind-service has two `/` characters. 

https://github.com/kubernetes/apimachinery/blob/master/pkg/util/validation/validation.go#L42-L71 Does a split on `/` and returns an error if len(parts) is not 1 or 2. 

@nschhina Maybe this is a docs issue then, because https://github.com/shopify/kubeaudit#override-labels seems to show examples in the format of audit.kubernetes.io/pod/... which appears to not be accepted by kubernetes.

**Comment by nschhina** [comment]
Thank you for reply @knisbet. I'll let @klautcomputing  answer for the first part, but I want to clarify what I meant- We use `kubeaudit.allow.capability.chown: "true"` style of label for capabilities and `audit.kubernetes.io/pod/...` style for pod overrides for rest of the checks.

**Comment by nschhina** [comment]
@knisbet  I just tried this on 
```apiVersion: v1
kind: Pod
metadata:
    name: Pod
    namespace: PodNamespace
    labels:
        audit.kubernetes.io/pod/allow-namespace-host-network: "some reason"
        audit.kubernetes.io/pod/allow-namespace-host-IPC: "some reason"
        audit.kubernetes.io/pod/allow-namespace-host-PID: "some reason"
spec:
    containers:
    - name: container
    hostPID: true
    hostIPC: true
    hostNetwork: true
``` 
And it didn't complain about two `/` for me

**Comment by nschhina** [comment]
Closing the issue as no further response was received
